### ISSUE TITLE
[libs/movex] Add Specs to the MemoryMovexStore Specs to test the `clearAll()` functionality

### ISSUE THREAD

**Issue by GabrielCTroia** [movesthatmatter/movex#46]
# Where

`libs/movex/src/lib/movex-store/MemoryMovexStore.spec.ts`

# What 

Add tests to assert the "clearAll()" functionality - the `clearAll()` method

# Why
To ensure the store works correctly

# Definition of done
- tests clearAll() with an empty store
- tests any newly created items aren't there after clearAll()
- tests clearAll() after items are updated
- all other tests are still passing

**Comment by RISHIKESHk07** [comment]
@GabrielCTroia  Hey i would like to contribute , don't have much knowledge about the issue could you guide about it

**Comment by GabrielCTroia** [comment]
Hi @RISHIKESHk07, sure!

So the clearAll() method simply removes all items in the store. Say, if you add 2 items and then call clearAll() the store should go back to being empty again!

Let me know if this makes sense! Thank you

**Comment by rohitverma007** [comment]
Added some tests but actually [the last one ](https://github.com/rohitverma007/movex/blob/0ec168dfd49d5612778bd4fb83313237df9ba666/libs/movex/src/lib/movex-store/MemoryMovexStore.spec.ts#L135-L142) fails, not sure if thats expected behaviour and the clearAll method might need change in functionality, happy to help fix it too:
```
  test('clearAll after multiple operations', async () => {
    const store = new MemoryMovexStore<{ counter: () => { count: number } }>();
    await store.create('counter:1', { count: 1 });
    await store.updateState('counter:1', (prev) => ({ count: prev.count + 1 }));
    await store.remove('counter:1');
    await store.create('counter:2', { count: 2 });
    await store.clearAll();
    expect(store.all()).toEqual({});
  });


```

**Comment by GabrielCTroia** [comment]
Hey @rohitverma007, thank you for this! I just replied in the PR as well. Yes, if you'd like to you're more than welcome to work on the functionality as well!

Thank you 🙏

**Comment by GabrielCTroia** [comment]
Closed in #188
